src/com/aca/vardanmanasaryan/homework1/management/Recordgadgets.java

```diff
@@ -0,0 +1,17 @@
+package com.aca.vardanmanasaryan.homework1.management;
+
+public class Recordgadgets extends Spygadgets implements Recordinggadget {
+    protected final String memory;
```

Comment:
can be item which doesn't have memory, so it shouldn't provide any memory property